Store usedAmount instead of remainingAmount

fleupold opened the pull request:
Closes #218

I added a small helper to compute the remaining amount to leave the change as minimal as possible.

### Test Plan

CI

fleupold committed:
Store usedAmount instead of remainingAmount

bh2smith reviewed (approved):
Would be happy to accept this asap and get all my open branches updated with this change.

fleupold committed:
Be more explicit about integer width (#281)

Using int instead of uint256 can be confusing (especially since we use different width generously in our smart contracts).

E.g. #277 noticed that we are downcasting a uint256 to u128 although the value it gets assigned to is of type uint.

This PR changes all uses of uint to uint256 and also removes the unnecessary downcast.

### Test Plan

CI

Closes #277

nlordell reviewed (approved):
Looks good, small nit (that can be ignored for now): all other private functions have doc strings, the new `getRemainingAmount()` doesn't

fleupold committed:
doc string
